plugins/modules/decompress.py

```diff
@@ -0,0 +1,215 @@
+author:
+  - Stanislav Shamilov (@shamilovstas)
+'''
+
+EXAMPLES = r'''
+- name: Decompress file /path/to/file.txt.gz into /path/to/file.txt (gz compression is used by default)
+  community.general.decompress:
+    src: /path/to/file.txt.gz
+    dest: /path/to/file.txt
+
+- name: Decompress file /path/to/file.txt.gz into /path/to/file.txt
+  community.general.decompress:
+    src: /path/to/file.txt.gz
+
+- name: Decompress file compressed with bzip2
+  community.general.decompress:
+    src: /path/to/file.txt.bz2
+    dest: /path/to/file.bz2
+    format: bz2
+
+- name: Decompress file and delete the compressed file afterwards
+  community.general.decompress:
+    src: /path/to/file.txt.gz
+    dest: /path/to/file.txt
+    remove: true
+'''
+
+RETURN = r'''
+dest:
+  description: Path to decompressed file
+  type: str
+  returned: success
+  sample: /path/to/file.txt
+'''
+
+import bz2
+import filecmp
+import gzip
+import os
+import shutil
+import tempfile
+
+from ansible.module_utils import six
+from ansible_collections.community.general.plugins.module_utils.mh.module_helper import ModuleHelper
+from ansible.module_utils.common.text.converters import to_native, to_bytes
+from ansible_collections.community.general.plugins.module_utils import deps
+
+with deps.declare("lzma"):
+    if six.PY3:
+        import lzma
+    else:
+        from backports import lzma
+
+
+def lzma_decompress(src):
+    return lzma.open(src, "rb")
+
+
+def bz2_decompress(src):
+    if six.PY3:
+        return bz2.open(src, "rb")
+    else:
+        return bz2.BZ2File(src, "rb")
+
+
+def gzip_decompress(src):
+    return gzip.open(src, "rb")
+
+
+def decompress(b_src, b_dest, handler):
+    with handler(b_src) as src_file:
+        with open(b_dest, "wb") as dest_file:
+            shutil.copyfileobj(src_file, dest_file)
+
+
+class Decompress(ModuleHelper):
+    destination_filename_template = "%s_decompressed"
+    use_old_vardict = False
+    output_params = 'dest'
+
+    module = dict(
+        argument_spec=dict(
+            src=dict(type='path', required=True),
+            dest=dict(type='path'),
+            format=dict(type='str', default='gz', choices=['gz', 'bz2', 'xz']),
+            remove=dict(type='bool', default=False)
+        ),
+        add_file_common_args=True,
+        supports_check_mode=True
+    )
+
+    def __init_module__(self):
+        self.handlers = {"gz": gzip_decompress, "bz2": bz2_decompress, "xz": lzma_decompress}
+        if self.vars.dest is None:
+            self.vars.dest = self.get_destination_filename()
+        deps.validate(self.module)
+        self.configure()
+
+    def configure(self):
+        b_dest = to_bytes(self.vars.dest, errors='surrogate_or_strict')
+        b_src = to_bytes(self.vars.src, errors='surrogate_or_strict')
+        if not os.path.exists(b_src):
+            msg = "Path does not exist: '%s'" % b_src
+            if self.vars.remove and os.path.exists(b_dest):
+                self.warn(msg)
```

Comment:
The idea was that it'd be helpful to let the user know that the file they are trying to decompress does not exist. IMO having a `src` non-existent is not something that should happen regularly. Initially I made it so that in case of non-existing `src`  the module would fail, but changed it after you commenting that it'd break idempodency. Since now the module finished successfully in this case, I thought that at least the user would be warned about this situation.

Reply:
This is not something I'm going to insist on, so I'll just delete the warn

Reply:
A warning is something that should only be used for very important events. For this case a (new) return value is better suited that can be queried by the user if they're interested in this case.